rbac/management/group/view.py

```diff
@@ -1126,14 +1141,41 @@ def remove_service_accounts(
 
             raise ValueError(f"Service account(s) {service_account_ids_diff} not found in the group '{group.name}'")
 
+        removed_service_accounts = []
         # Remove service accounts from the group.
         with transaction.atomic():
             for service_account in valid_service_accounts:
                 group.principals.remove(service_account)
+                removed_service_accounts.append(service_account)
 
         logger.info(
             f"[Request_id:{request_id}] {valid_service_account_ids} "
             f"removed from group {group.name} for org id {org_id}."
         )
         for username in service_accounts:
             group_principal_change_notification_handler(self.request.user, group, username, "removed")
+
+        return removed_service_accounts
+
+    def _get_add_principals_queryset(self):
+        """Obtain queryset for requesting user based on access to add principals."""
+        return Group.objects.filter(tenant=self.request.tenant).select_for_update()
```

Comment:
Do we need to port anything else from `get_group_queryset` here?

There are various filters that take into account user access and query params. I'm not sure if those are relevant here, but we did end up porting some of that over for the Role view at least.

This is maybe one cost of locking – if we don't really need to, we could consider keeping the same query and we wouldn't have to run a second query, either.

Reply:
yes I am going to remove locking as I mentioned in previous commit.